packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -85,14 +85,14 @@ export class Metric {
   /**
    * Grant permissions to the given identity to write metrics.
    *
-   * @param identity The IAM identity to give permissions to.
+   * @param grantee The IAM identity to give permissions to.
    */
-  public static grantPutMetricData(identity?: iam.IPrincipal) {
-    if (!identity) { return; }
-
-    identity.addToPolicy(new iam.PolicyStatement()
-      .addAllResources()
-      .addAction("cloudwatch:PutMetricData"));
+  public static grantPutMetricData(grantee: iam.IGrantable): iam.Grant {
+    return iam.Grant.onPrincipal({
```

Comment:
"grant on principal" sounds weird. I understand that this is coming from "add a grant on the principal's policy" and so it can work with "add a grant on the principal policy and resource policy", but maybe we can do better... `Grant.addToPrincipal` and `Grant.addToPrincipalAndResource`? I think the "add" is missing to indicate that there's a mutation going on.

Reply:
Good call.